lib/semantics/resolve-names.cc

```diff
@@ -2940,37 +2948,4 @@ static GenericSpec MapGenericSpec(const parser::GenericSpec &genericSpec) {
       genericSpec.u);
 }
 
-static void PutIndent(std::ostream &os, int indent) {
-  for (int i = 0; i < indent; ++i) {
-    os << "  ";
-  }
-}
-
-static void DumpSymbols(std::ostream &os, const Scope &scope, int indent = 0) {
-  PutIndent(os, indent);
-  os << Scope::EnumToString(scope.kind()) << " scope:";
-  if (const auto *symbol{scope.symbol()}) {
-    os << ' ' << symbol->name().ToString();
-  }
-  os << '\n';
-  ++indent;
-  for (const auto &pair : scope) {
-    const auto &symbol{*pair.second};
-    PutIndent(os, indent);
-    os << symbol << '\n';
-    if (const auto *details{symbol.detailsIf<GenericDetails>()}) {
-      if (const auto &type{details->derivedType()}) {
-        PutIndent(os, indent);
-        os << *type << '\n';
-      }
-    }
-  }
-  for (const auto &child : scope.children()) {
-    DumpSymbols(os, child, indent);
-  }
-  --indent;
-}
-
-void DumpSymbols(std::ostream &os) { DumpSymbols(os, Scope::globalScope); }
```

Comment:
I assume this is just debugging code?

Reply:
DumpSymbols() is called when -fdebug-dump-symbols is specified.